sockeye/inference.py

```diff
@@ -54,6 +54,10 @@ def __init__(self,
                  beam_size: int,
                  checkpoint: Optional[int] = None,
                  softmax_temperature: Optional[float] = None):
+        self.model_version = utils.load_version(os.path.join(model_folder, C.VERSION_NAME))
```

Comment:
I had an old model that failed because it didn't have this max_len attribute, rather than failing because of the version check. So I guess it makes sense to do the version check as the first check, before accessing any config attributes.